A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2558,19 +2570,42 @@ class FMCMainDisplay extends BaseAirliners {
         return false;
     }
 
-    setPerfApprTransAlt(s) {
-        if (!/^\d{4,5}$/.test(s)) {
+    trySetPerfApprTransAlt(s) {
+        if (s === FMCMainDisplay.clrValue) {
+            this.transitionArrivalAltitude = NaN;
+            this.transitionArrivalAltitudeIsPilotEntered = false;
+            SimVar.SetSimVarValue("L:AIRLINER_APPR_TRANS_ALT", "Number", 0);
+            return true;
+        }
+
+        let value = parseInt(s);
+        if (!isFinite(value) || !/^\d{4,5}$/.test(s)) {
             this.addNewMessage(NXSystemMessages.formatError);
             return false;
         }
-        const v = parseInt(s);
-        if (isFinite(v) && v > 0) {
-            this.perfApprTransAlt = v;
-            SimVar.SetSimVarValue("L:AIRLINER_APPR_TRANS_ALT", "Number", v);
-            return true;
+        value = Math.round(value / 10) * 10;
+        if (value < 1000 || value > 45000) {
+            this.addNewMessage(NXSystemMessages.entryOutOfRange);
+            return false;
         }
-        this.addNewMessage(NXSystemMessages.notAllowed);
-        return false;
+        this.transitionArrivalAltitude = value;
+        this.transitionArrivalAltitudeIsPilotEntered = true;
+        SimVar.SetSimVarValue("L:AIRLINER_APPR_TRANS_ALT", "Number", value);
+        return true;
+    }
+
+    async getArrivalTransitionAltitude(arrivalICAO) {
+        await NXApi.getAirport(arrivalICAO)
```

Comment:
**question**: Why do you use `async` and `await` here? What is your intended effect?

Reply:
Same as above